app/js/addCategories.ts

```diff
@@ -0,0 +1,44 @@
+async function addCategory():Promise<void> {
+    let categoriesObject = await getData('/category');
+    const responseMessage = document.querySelector('#addedCategoryConfirmation');
+    document.querySelector<HTMLFormElement>('.categoriesForm').addEventListener('submit', (e) => {
+        e.preventDefault();
+        let newCategory: string = document.querySelector<HTMLFormElement>('#categoryName').value;
+        if (newCategory.length > 255) {
+            responseMessage.textContent = 'Error - Category Name Exceeds 255 characters.';
+            responseMessage.classList.add('alert-danger');
+            responseMessage.classList.remove('alert-success');
+        } else {
+            let listOfCategories = [];
+            categoriesObject.data.forEach((category) => {
+                listOfCategories.push(category.name);
+            })
+            if (!listOfCategories.includes(newCategory)) {
+                let data: object = {"name": newCategory};
+                let dataToSend: FormData = jsonToFormData(data);
+                document.querySelector<HTMLFormElement>('#categoryName').value = '';
+
+                sendData(dataToSend, 'category').then((data) => {
+
+                    if (data.success) {
+                        responseMessage.textContent = 'Success! Category added';
+                        responseMessage.classList.remove('alert-danger');
+                        responseMessage.classList.add('alert-success');
```

Comment:
when you add a category, the success box never disappears. it should have a timeout on it to fade out or hide or something